src/trace-plugin-loader.js

```diff
@@ -121,15 +122,18 @@ function activate(agent) {
 
       for (var file in patchSet) {
         var patch = patchSet[file];
+        var loadPath = moduleRoot ? path.join(moduleRoot, patch.file) : patch.file;
         if (!patch.module) {
-          var loadPath = moduleRoot ? path.join(moduleRoot, patch.file) : patch.file;
           patch.module = originalModuleLoad(loadPath, module, false);
         }
         if (patch.patch) {
           patch.patch(patch.module, instrumentation.api);
         }
         if (patch.intercept) {
           patch.module = patch.intercept(patch.module, instrumentation.api);
+          intercepts[loadPath] = {
+            interceptedValue: patch.module
```

Comment:
If the intercepted value is falsey then it becomes more complicated to check whether the value was replaced or not. Ex. returning `undefined` from the intercept function (though, not sure why one would do that).